framework/src/wifi_manager/wifi_manager.c

```diff
@@ -1586,3 +1608,43 @@ wifi_manager_result_e wifi_manager_get_stats(wifi_manager_stats_s *stats)
 	}
 	return wret;
 }
+
+wifi_manager_result_e wifi_manager_register_cb(wifi_manager_cb_s *wmcb)
+{
+	wifi_manager_result_e res = WIFI_MANAGER_FAIL;
+
+	LOCK_WIFIMGR;
+	// g_manager_info.cb[0] is assigned to the callback which is registered by wifi_manager_init
+	if (g_manager_info.cb[0] == NULL) {
+		UNLOCK_WIFIMGR;
+		return WIFI_MANAGER_DEINITIALIZED;
+	}
+	int i = 1;
```

Comment:
declaration could be added at the start of wifi_manager_register_cb API with rest of the declarations
Initialization in the for loop